### DISCUSSION TITLE
How we can reduce latency between Primary Postgres cluster and Replica Cluster in CNPG for Kubernetes environment?

### DISCUSSION THREAD

**Discussion by oberai07** [cloudnative-pg/cloudnative-pg#2633] [General]
Currently, we have a setup running with Primary and multiple Replica clusters. We are also using S3 object storage to store  WAL archives and base backups. Our Replica clusters are pointing to this storage to replicate the data but we have a data latency of approximately ~5 minutes and data is not getting replicated immediately.  

Is there any way to reduce the latency or could you please suggest parameters to tune it?

**Comment by gbartolini** [comment]
You can do it if you also set streaming connection between the replica cluster and the source cluster, namely `connectionParameters`. See: https://cloudnative-pg.io/documentation/current/replica_cluster/
